index.rst

```diff
@@ -1,61 +1,365 @@
+    The identity management system need not understand the quotas, but it should be able to sum multiple quotas under the same label.
+
+IDM-1201
+    The user must be able to view all of their existing quotas.
+
+IDM-1202
+    The user must be able to request a new quota grant.
+    That request should be routed to some approval process by a manager of the relevant resource, who can then grant or deny the request via the identity management web interface.
+
+IDM-1203
+    Quota grants may expire.
+    The user must be notified via email of pending quota grant expirations.
+
+Administration
+--------------
+
+IDM-1300
+    Administrators of the Science Platform must be able to modify any of the user's metadata on behalf of the user.
+
+IDM-1301
+    Administrators of the Science Platform must be able to set and change expiration dates on accounts.
+
+IDM-1302
+    Administrators of the Science Platform must be able to approve a pending change of email address even if the user has not responded to the challenge.
+
+IDM-1303
+    Administrators of the Science Platform must be able to create, revoke, and change the expiration dates on quota grants.
+
+IDM-1304
+    Administrators must be able to impersonate a user and see the same thing that a user would see in the user metadata interface.
+
+IDM-1305
+    Administrators must be able to impersonate a user to other Science Platform services so that an administrator can debug issues that only affect a single user.
+
+Logging
+-------
+
+IDM-1400
+    All changes to account metadata must be logged.
+    If the changes were made by an administrator instead of the user, this must be clearly indicated in the logs.
+
+IDM-1401
+    All changes to quotas associated with users must be logged.
+
+IDM-1402
+    Users must be able to see a history of all of their quota changes.
+
+Groups
+======
+
+Management
+----------
+
+IDM-2000
+    Users may be members of zero or more groups.
+
+IDM-2001
+    Groups can be configured to control membership based on attributes provided by the identity provider.
+    Membership in those groups must be tied to affiliation information from specific identity providers and dynamically adjusted if an authentication from that identity provider stops returning the same metadata.
+    Multiple identity providers may provide access to the same group.
+    In this case, the membership should only be withdrawn if all those identity providers stop providing the relevant information.
+    It must be possible to periodically force users to authenticate with an attribute-providing identity provider to reconfirm access to their groups, similar to IDM-0009.
+
+IDM-2002
+    Users must be able to create their own groups.
+    The owner of the group must then be able to add and remove members as they wish.
+    Owners must also be able to add additional group owners who can then also control membership in the group.
```

Comment:
You may not want to require this, but I like the pattern from AFS where administrative access to a group can be controlled by membership in another group (a "group of owners").  This can be particularly useful where a complex collaboration (e.g., DESC) has several groups it controls: ordinary membership in the collaboration, members with write access to production resources of the collaboration, etc.

Having admin control of these groups associated with another group helps out when, over time, the "super-users" for the collaboration change.  You don't then have to go into each separate group and change the owner list.

Reply:
I'll go ahead and add ownership of groups by other groups. We can prioritize whether to implement it based on how difficult we will be. If we go with COmanage, Grouper supports this out of the box.